src/DependencyManagement/DependencyManager.cs

```diff
@@ -343,20 +390,43 @@ internal void SetDependenciesDestinationPath(string path)
         }
 
         /// <summary>
-        /// Validates that the given major version is less or equal to the latest supported major version.
+        /// Gets the latest module version from the PSGallery for the given module name and major version.
         /// </summary>
-        private void ValidateModuleMajorVersion(string moduleName, string majorVersion, string latestVersion)
+        internal string GetModuleLatestSupportedVersion(string moduleName, string majorVersion)
         {
-            // A Version object cannot be created with a single digit so add a '.0' to it.
-            var requestedVersion = new Version($"{majorVersion}.0");
-            var latestSupportedVersion = new Version(latestVersion);
+            string latestVersion = null;
+
+            string errorDetails = null;
+            bool throwException = false;
 
-            if (requestedVersion.Major > latestSupportedVersion.Major)
+            try
             {
-                // The requested major version is greater than the latest major supported version.
-                var errorMsg = string.Format(PowerShellWorkerStrings.InvalidModuleMajorVersion, moduleName, majorVersion);
-                throw new ArgumentException(errorMsg);
+                latestVersion = GetLatestModuleVersionFromThePSGallery(moduleName, majorVersion);
+            }
+            catch (Exception e)
+            {
+                throwException = true;
+
+                if (!string.IsNullOrEmpty(e.Message))
+                {
+                    errorDetails = string.Format(PowerShellWorkerStrings.ErrorDetails, e.Message.ToString());
+                }
             }
+
+            // If we could not find the latest module version error out.
+            if (string.IsNullOrEmpty(latestVersion) || throwException)
+            {
+                if (string.IsNullOrEmpty(errorDetails))
```

Comment:
You could probably write just `errorDetails = errorDetails ?? string.Empty` instead of these four lines, or (perhaps even better) pass `errorDetails ?? string.Empty` to string.Format directly.

Reply:
Thanks for the suggestion. Code updated.